### PR TITLE
Speeds up ghcn_splitvars

### LINKED ISSUE #352: ghcnd_splitvars can be faster
All the data manipulations performed by `ghcnd_splitvars()` can be very slow.  In my machine, processing one station id can take almost 2 seconds:

``` r
library(rnoaa)

station <- ghcnd_stations()
station <- ghcnd(station$id[1])

system.time(ghcnd_splitvars(station))
#>    user  system elapsed 
#>   1.941   0.004   1.948
```

The culprit is [all these](https://github.com/ropensci/rnoaa/blob/master/R/ghcnd.R#L128) dplyr manipulations and `tidyr::gather()` calls that are somewhat redundant. [I experimented a little](https://github.com/eliocamp/rnoaa/blob/dt-ghcnd_splitvars/R/ghcnd.R#L126) using `data.table::melt()` and got dramatically better performance:

``` r
library(rnoaa) # Using eliocamp/rnoaa@dt-ghcnd_splitvars

station <- ghcnd_stations()
station <- ghcnd(station$id[1])

bench::mark(new = ghcnd_splitvars(station),
            old = rnoaa:::ghcnd_splitvars2(station))
#> Warning: Some expressions had a GC in every iteration; so filtering is disabled.
#> # A tibble: 2 x 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 new         58.13ms  60.75ms    14.7      4.64MB     1.84
#> 2 old           1.93s    1.93s     0.519   12.16MB     2.08
```

<sup>Created on 2020-06-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

So, from 2s to less than 60ms! 

If you like, I can create a PR with the change (minus `ghcnd_splitvars2`, of course).

### PR DESCRIPTION
**Pull request by eliocamp**
Changes a lot of dplyr data manipulation into a `data.table::melt()`. This makes `ghcn_splitvars()` considerably faster. 

Closes #352

### CODE REVIEW

**Comment by sckott** [comment]
can you incorporate the latest commit from master to your branch - then I'll take a look

File: R/ghcnd.R [hunk 1]
```diff
@@ -229,3 +201,5 @@ str_extract_ <- function(string, pattern) {
 str_extract_all_ <- function(string, pattern) {
   regmatches(string, gregexpr(pattern, string))
 }
+
+.datatable.aware = TRUE
```
Comment (sckott) [inline R/ghcnd.R:205]: what does `.datatable.aware`  do?

**Comment by eliocamp** [comment]
> can you incorporate the latest commit from master to your branch - then I'll take a look

Sorry, I'm still a git novice. The git graph is showing me that this branch is based on current master. What am I missing?

image

**Comment by sckott** [comment]
ah nevermind, was looking at rebase, but we can merge

**Pull request merged**